projects/common-platform-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/PersonService.kt

```diff
@@ -75,8 +81,35 @@ class PersonService(
 
             val savedEquality = equalityRepository.save(equality)
 
-            val savedAddress =
-                defendant.personDefendant.personDetails.address.takeIf { it.containsInformation() }?.let {
+            val addressInfo = defendant.personDefendant.personDetails.address
+            val osPlacesResponse = addressInfo?.takeIf { it.containsInformation() && !it.postcode.isNullOrBlank() }
+                ?.let { findAddressByFreeText(it) }
+
+            val deliveryPointAddress = osPlacesResponse?.results?.firstOrNull()?.dpa
+            val isValidMatch = deliveryPointAddress?.match?.let { it >= 0.6 } ?: false
+
+            val savedAddress = if (deliveryPointAddress != null && isValidMatch) {
+                insertAddress(
+                    PersonAddress(
+                        id = null,
+                        start = LocalDate.now(),
+                        status = referenceDataRepository.mainAddressStatus(),
+                        person = savedPerson,
+                        type = referenceDataRepository.awaitingAssessmentAddressType(),
+                        postcode = deliveryPointAddress.postcode,
+                        notes = deliveryPointAddress.address,
+                        buildingName = listOfNotNull(
+                            deliveryPointAddress.subBuildingName,
+                            deliveryPointAddress.buildingName
+                        ).joinToString(" "),
+                        addressNumber = deliveryPointAddress.buildingNumber?.toString(),
+                        streetName = deliveryPointAddress.thoroughfareName,
+                        town = deliveryPointAddress.postTown,
+                        district = deliveryPointAddress.localCustodianCodeDescription
+                    )
+                )
+            } else {
+                addressInfo?.takeIf { it.containsInformation() }?.let {
```

Comment:
Are we planning to break this down into the specific address fields here or leaving it in the notes?

Reply:
Discussed this and have agreed a mapping rather than putting all the info in the notes fields